src/components/productCardShowMoreButton.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import Image from "next/image";
+
+type ContentProps = {
+  backOfCardImage?: string[];
+  image: string[];
+};
+
+type ProductCardProps = {
+  content: ContentProps;
+  isCardFlipped: boolean;
+  changeImage?: (newImage?: string) => void;
+  toggleIsFlippedCard: () => void;
+};
+
+const backOfCardImageIsValid = (ContentProps: ContentProps) => {
+  if (ContentProps.backOfCardImage !== undefined) {
+    return ContentProps.backOfCardImage[0];
+  }
+};
+
+export default function productCardShowMoreButton({
+  content,
+  isCardFlipped,
+  changeImage,
+  toggleIsFlippedCard,
+}: ProductCardProps) {
+  const frontProductImage = content.image[0];
+  const backProductImage = backOfCardImageIsValid(content);
+
+  return (
+    <div className="z-50">
+      <button
+        className="group-hover:visible invisible flex flex-row absolute top-0 right-0 py-1 px-2 text-xs bg-gray-300"
+        onClick={() => {
+          toggleIsFlippedCard();
+          changeImage?.(isCardFlipped ? backProductImage : frontProductImage);
```

Comment:
what is going on here? This seems super odd. if changeImage function exists, you're calling a property on the function? Or is that invocation of the function?

Reply:
I'm invoking the function here. The issue was, that backProductImage was  possibly undefined and typescript would complain. So I had to add the "?." so typescript knows that the the function is aware it's possibly undefined.

Reply:
This syntax is highly unusual. Does this work currently? At no point do I see it being invoked.

if I'm reading this correctly it evaluates out to this. changeImage.backProductImage or changeImage.frontProductImage if at all. The parentheses after the period is super odd

Reply:
it works. its called optional chaining. https://marketsplash.com/tutorials/typescript/how-to-fix-object-is-possibly-undefined-typescript/#link5

Reply:
I know what optional chaining is, what is the .(isCardFlipped ? backProductImage : frontProductImage) syntax? Wouldn't that evaluate to the function itself as opposed to a function call?

Reply:
From my understanding when  toggleIsFlippedCard(); is called it will set isCardFlipped to true of false, which would then determine which string will be returned. I suppose yes, it would be the function itself as the logic is within the parameters of changeImage. Is there a clear pro and con of having it this way, vs as function that i would invoke inside of changeImage.